File: ContainerWithMostWater.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
#include<stdbool.h>
#include<math.h>
int maxArea(int* height, int heightSize)
{
	int l = 0, r = heightSize - 1, max = 0;
	while (l < r)
	{
		int temp = height[l] - height[r];
		if (max < (r - l) * (temp < 0 ? height[l] : height[r]))
			max = (r - l) * (temp < 0 ? height[l] : height[r]);
		if (temp < 0)
			++ l;
		else
			-- r;
	}
	return max;
}
int main()
{
	FILE *fin = fopen("oo.xx", "r");
	FILE *fout = fopen("xx.oo", "w");
	int n, a[100001];
	fscanf(fin, "%d", &n);
	for (int i = 0; i < n; ++ i)
		fscanf(fin, "%d", a + i);
	int out = maxArea(a, n);
	printf("%d\n", out);
	fprintf(fout, "%d\n", out);
	return 0;
}

File: ContainerWithMostWater.py
from typing import List
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r, out = 0, len(height) - 1, 0
        while (l < r):
            temp = height[l] - height[r]
            if out < (r - l) * (height[l] if temp < 0 else height[r]):
                out = (r - l) * (height[l] if temp < 0 else height[r]) 
            if temp < 0:
                l += 1
            else:
                r -= 1 
        return out

fin = open("oo.xx", "r")
fout = open("xx.oo", "w")
a = [int(x) for x in fin.readline()[:-1].split(',')]
out = Solution().maxArea(a)
print(out)
fout.write(str(out))

